pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -407,6 +407,15 @@ fi
 
 tmpDir=$(mktemp -t -d nixos-rebuild.XXXXXX)
 
+# macOS sets the TMPDIR to /var/folders/ab/abcdef.../A as a per-user setting
+# to improve security. However, this does not play well with SSH control paths:
+# unix_listener: path "/var/folders/ab/abcdef.../T/nixos-rebuild...." too long
+# for Unix domain socket
+# Unset TMPDIR in this case
+if [[ $tmpDir = /var/folders/* ]]; then
```

Comment:
Instead of matching the folder, I think you should check the length of `$tmpDir`

Reply:
This feels somewhat imprecise because the tmpDir plus `ssh-%n` together are too long when it fails, so this is really just some heuristic now. but i changed it to check the length now.